src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php

```diff
@@ -83,6 +84,13 @@ protected function displaySingleMessage(Envelope $envelope, SymfonyStyle $io)
 
         $io->table([], $rows);
 
+        /** @var RedeliveryStamp $redeliveryStamp */
+        $redeliveryStamp = $envelope->last(RedeliveryStamp::class);
+        if (null !== $redeliveryStamp) {
+            $io->writeln(sprintf(' This message has already been tried by the failure transport %d time%s before.', $redeliveryStamp->getRetryCount(), 1 !== $redeliveryStamp->getRetryCount() ? 's' : ''));
```

Comment:
I don't understand the meaning of this. Is that the amount of retries before it landed in the failure transport? Or the amount of times it went into the failure transport, i.e. how often you manually retried it? The text could be improved.

Reply:
I guess it cannot be the amout of times in went into the failure transport as that cannot happen as I pointed out below.

Reply:
As I mentioned below, even the failure transport acts like a normal transport: if you try a message on the failure transport once and it fails, it will actually go *back* into the failure transport to be retried again, until it hits the max retries (default 3). This is telling you:

> Hey! FYI! This is not a newly failed message - you've already retried this N times in
> the past on the failure transport it it failed.

Reply:
I have an idea: The SendFailedMessageToFailureTransportListener adds a new RedeliveryStamp with count 0. But the previous RedeliveryStamp should still be there. So we could also display the amount of retries it had before going in the failure transport. I think this is useful since you can have different retry strategies.

And the retries in the failure transport could IMO only be displayed if greater than zero. Currently it would always display the message even with `0 times`.

Reply:
I love that idea! With some slight modifications, here's the final result:

<img width="1263" alt="Screen Shot 2019-05-09 at 10 50 13 AM" src="https://user-images.githubusercontent.com/121003/57463217-5e7d6a00-7248-11e9-8542-2ecf1d6d6f6d.png">

I actually don't show the "times retried" because it's not needed when you can see the history piece-by-piece. And when the retry count is 0 for the failure transport, I still print that `RedeliveryStamp` because it adds some nice history.

Reply:
I like the history approach. That gives me another idea :)
Each failure can have a different reason/exception. So I suggest to to add the exception to the RedeliveryStamp. This you can display the error for each failure. And you don't even need the exception in the SentToFailureTransportStamp anymore.

Reply:
Ah! I really like that idea too!

Buuuut, in addition to the error message, we also store the flattened exception on `SentToFailureTransportStamp`. I'm not sure how I feel about adding all of that to each retry - it would make the messages pretty big (is that really a problem?).

Reply:
I don't think that the message size is a problem. And if it was, we could for example only add the exception if the error message is different than the last one before (which otherwise likely means the exception trace is the same one). Or we compare the FlattenException directly with the previous one.

Reply:
Done! The result is exactly the same (for now, I'm not trying to show the exception message from *every* `RedeliveryStamp`) - but the error information is now "packaged" with all the RedeliveryStamps.